iceoryx2-pal/posix/src/macos/mman.rs

```diff
@@ -74,23 +78,94 @@ unsafe fn shm_file_path(name: *const char, suffix: &[u8]) -> [u8; MAX_PATH_LENGT
     state_file_path
 }
 
+unsafe fn get_real_shm_name(name: *const char) -> Option<[u8; SHM_MAX_NAME_LEN]> {
+    let shm_file_path = shm_file_path(name, SHM_STATE_SUFFIX);
+    let shm_state_fd = open_with_mode(shm_file_path.as_ptr().cast(), O_RDONLY, 0);
+    if shm_state_fd == -1 {
+        return None;
+    }
+
+    let mut buffer = [0u8; SHM_MAX_NAME_LEN];
+
+    if read(shm_state_fd, buffer.as_mut_ptr().cast(), SHM_MAX_NAME_LEN) <= 0 {
+        close(shm_state_fd);
+        return None;
+    }
+
+    close(shm_state_fd);
+    Some(buffer)
+}
+
+unsafe fn write_real_shm_name(name: *const char, buffer: &[u8]) -> bool {
```

Comment:
This `char` is not the `std::char` but an alias to a `u8`, right? I would urge to rename it to `c_char` or something similar since it is quite confusing when reading the code.

Reply:
nope `char` is an alias to `core::ffi::c_char` but I get your point, I will create an issue

Reply:
See: https://github.com/eclipse-iceoryx/iceoryx2/issues/54